buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -116,6 +122,15 @@ class ClusterFormationTasks {
         project.dependencies.add(configuration.name, "org.elasticsearch.distribution.${distro}:elasticsearch:${elasticsearchVersion}@${packaging}")
     }
 
+    /** Adds a dependency on a different version of the given plugin, which will be retrieved using gradle's dependency resolution */
+    static void configureBwcPluginDependency(Project project, Project pluginProject, Configuration configuration, String elasticsearchVersion) {
+        if (pluginProject.plugins.hasPlugin(PluginBuildPlugin) == false) {
+            throw new GradleException("Cannot use project ${pluginProject.path} which is not an esplugin")
```

Comment:
Could it be a more descriptive message? Something like "Cannot add plugin [${pluginProject.path}] in version [${elasticsearchVersion}] to project's [${project.path}] dependency: the project plugin is not an esplugin"